File: truth_set/javascript/structure/organisation/consistency/supermarket.js
var UserService = /** @class */ (function () {
    function UserService() {
    }
    UserService.prototype.fetchUserProfile = function (userId) {
        return new Promise(function (resolve) {
            setTimeout(function () {
                resolve("User Profile for ID: ".concat(userId));
            }, 1000);
        });
    };
    UserService.prototype.checkUserEligibility = function (age) {
        return age >= 18;
    };
    return UserService;
}());
var ProductService = /** @class */ (function () {
    function ProductService() {
    }
    ProductService.prototype.retrieveProductDetail = function (productId) {
        var products = {
            101: "Laptop",
            102: "Smartphone",
            103: "Headphones",
        };
        return products[productId] || "Product Not Found";
    };
    ProductService.prototype.validateProductStock = function (stock) {
        return stock > 0;
    };
    return ProductService;
}());
var OrderService = /** @class */ (function () {
    function OrderService() {
    }
    OrderService.prototype.getOrderInfo = function (orderId) {
        var orders = {
            5001: "Order #5001: Shipped",
            5002: "Order #5002: Processing",
            5003: "Order #5003: Delivered",
        };
        return orders[orderId] || "Order Not Found";
    };
    OrderService.prototype.isOrderComplete = function (completed) {
        return completed;
    };
    return OrderService;
}());
var userService = new UserService();
userService.fetchUserProfile(1).then(function (profile) { return console.log(profile); });
console.log(userService.checkUserEligibility(20));
var productService = new ProductService();
console.log(productService.retrieveProductDetail(101));
console.log(productService.validateProductStock(0));
var orderService = new OrderService();
console.log(orderService.getOrderInfo(5002));
console.log(orderService.isOrderComplete(true));

File: truth_set/python/structure/organisation/consistency/supermarket.py
import asyncio
from typing import Dict

class UserService:
    def __init__(self):
        pass
        
    async def fetch_user_profile(self, user_id: int) -> str:
        await asyncio.sleep(1)
        return f"User Profile for ID: {user_id}"
        
    def check_user_eligibility(self, age: int) -> bool:
        return age >= 18

class ProductService:
    def __init__(self):
        pass
        
    def retrieve_product_detail(self, product_id: int) -> str:
        products: Dict[int, str] = {
            101: "Laptop",
            102: "Smartphone", 
            103: "Headphones"
        }
        return products.get(product_id, "Product Not Found")
        
    def validate_product_stock(self, stock: int) -> bool:
        return stock > 0

class OrderService:
    def __init__(self):
        pass
        
    def get_order_info(self, order_id: int) -> str:
        orders: Dict[int, str] = {
            5001: "Order #5001: Shipped",
            5002: "Order #5002: Processing",
            5003: "Order #5003: Delivered"
        }
        return orders.get(order_id, "Order Not Found")
        
    def is_order_complete(self, completed: bool) -> bool:
        return completed

async def main():
    user_service = UserService()
    profile = await user_service.fetch_user_profile(1)
    print(profile)
    print(user_service.check_user_eligibility(20))

    product_service = ProductService()
    print(product_service.retrieve_product_detail(101))
    print(product_service.validate_product_stock(0))

    order_service = OrderService()
    print(order_service.get_order_info(5002))
    print(order_service.is_order_complete(True))

if __name__ == "__main__":
    asyncio.run(main())